cms/djangoapps/contentstore/rest_api/v1/tests/test_views.py

```diff
@@ -0,0 +1,93 @@
+from student.tests.factories import UserFactory
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+
+# from cms.djangoapps.contenstore.tests.test_course_settings import get_url
+class ProctoringExamSettingsGetTests(SharedModuleStoreTestCase, APITestCase):
+    """ Tests for proctored exam settings GETs """
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.course_key = CourseKey.from_string('course-v1:edX+ToyX+Toy_Course')
+        cls.other_course_key = CourseKey.from_string('course-v1:edX+ToyX_Other_Course+Toy_Course')
+        cls.course = cls.create_course_from_course_key(cls.course_key)
+        cls.other_course = cls.create_course_from_course_key(cls.other_course_key)
+        cls.password = 'password'
+        cls.student = UserFactory.create(username='student', password=cls.password)
+        cls.global_staff = GlobalStaffFactory(username='global-staff', password=cls.password)
+        cls.course_instructor = InstructorFactory(
+            username='instructor',
+            password=cls.password,
+            course_key=cls.course.id,
+        )
+        cls.other_course_instructor = InstructorFactory(
+            username='other-course-instructor',
+            password=cls.password,
+            course_key=cls.other_course.id,
+        )
+
+    def tearDown(self):
+        super().tearDown()
+        self.client.logout()
+
+    @staticmethod
+    def create_course_from_course_key(course_key):
+        return CourseFactory.create(
+            org=course_key.org,
+            course=course_key.course,
+            run=course_key.run
+        )
+
+    def get_url(self, course_key):
+        return reverse(
+            'cms.djangoapps.contentstore:v1:proctored_exam_settings',
+            kwargs={'course_id': course_key}
+        )
+
+    def test_403_if_student(self):
+        self.client.login(username=self.student.username, password=self.password)
+        url = self.get_url(self.course.id)
+        response = self.client.get(url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_403_if_instructor_in_another_course(self):
+        self.client.login(
+            username=self.other_course_instructor.username,
+            password=self.password
+        )
+        url = self.get_url(self.course.id)
+        response = self.client.get(url)
+        assert response.status_code == status.HTTP_403_FORBIDDEN
+
+    def test_200_global_staff(self):
+        self.client.login(username=self.global_staff.username, password=self.password)
+        url = self.get_url(self.course.id)
+        response = self.client.get(url)
+        assert response.status_code == status.HTTP_200_OK
+
+    def test_200_course_instructor(self):
+        self.client.login(username=self.course_instructor.username, password=self.password)
```

Comment:
My running theory is that this is failing because of the `permissions.JWT_RESTRICTED_APPLICATION_OR_USER_ACCESS` permissions class, which is defined [here](https://github.com/edx/edx-drf-extensions/blob/8303f129e0b770705cc669e60daf174a6e37b0b9/edx_rest_framework_extensions/permissions.py#L171). Since we're not using JWT authentication here, we care about the `_NOT_JWT_RESTRICTED_PERMISSIONS` condition, defined [here](https://github.com/edx/edx-drf-extensions/blob/8303f129e0b770705cc669e60daf174a6e37b0b9/edx_rest_framework_extensions/permissions.py#L164). It seems that the requesting user either needs to be staff or the username of the requesting user needs to be passed as a query parameter in the URL. I don't see other tests for DRF endpoints needing to include this query parameter, so I'm confused.

In fact, in each test here besides `test_200_global_staff`, the view isn't ever invoked, because it's returning out with a 403 due to this check.